### PR TITLE
Oi4 340

### PR DESCRIPTION
**Pull request by RiccardoScuderi-EH**

### CODE REVIEW

File: packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts [hunk 1]
```diff
@@ -36,6 +36,13 @@ export class OPCUABuilder {
     this.topicRex = new RegExp(topicPathSchemaJson.pattern);
   }
 
+  private setAtomarLastMessageId(opcUaDataMessage: IOPCUANetworkMessage | IOPCUAMetaData) {
+    for(let counter = 0; this.lastMessageId === opcUaDataMessage.MessageId; counter++) {
```
Comment (Weinschenk) [inline packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts]: Loop solution really a good way to solve the issue 👍🏻

File: packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts [hunk 2]
```diff
@@ -36,6 +36,13 @@ export class OPCUABuilder {
     this.topicRex = new RegExp(topicPathSchemaJson.pattern);
   }
 
+  private setAtomarLastMessageId(opcUaDataMessage: IOPCUANetworkMessage | IOPCUAMetaData) {
```
Comment (Weinschenk) [inline packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts]: Setting values of a provided parameter is a side effect. Side effects are a big source for failures. Better return the value and let the caller set the value

File: packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts [hunk 3]
```diff
@@ -36,6 +36,13 @@ export class OPCUABuilder {
...
+    for(let counter = 0; this.lastMessageId === opcUaDataMessage.MessageId; counter++) {
+        opcUaDataMessage.MessageId = `${counter}${opcUaDataMessage.MessageId}`;
+        return;
+    }
+    this.lastMessageId =  opcUaDataMessage.MessageId;
```
Comment (Weinschenk) [inline packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts]: Return the found ID

File: packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts [hunk 4]
```diff
@@ -147,11 +150,7 @@ export class OPCUABuilder {
       correlationId: correlationId,
       MetaData: opcUaMetaDataPayload,
     };
-    if (this.lastMessageId === opcUaMetaDataMessage.MessageId) {
-      opcUaMetaDataMessage.MessageId = `OverFlow${opcUaMetaDataMessage.MessageId}`;
-    } else {
-      this.lastMessageId = opcUaMetaDataMessage.MessageId;
-    }
+    this.setAtomarLastMessageId(opcUaMetaDataMessage);
```
Comment (Weinschenk) [inline packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts]: use something like 
```suggestion
    const messageId =  getUniqueMessageId(`${Date.now().toString()}-${this.publisherId}`)
```
use it in before the initialization of the  IOPCUAMetaData in line 143 and pass the value directly. This avoid the changing of the value
Furthermore

**Review by Weinschenk — CHANGES_REQUESTED** [review]

File: packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts (hunk 2 above)
Comment (RiccardoScuderi-EH) [reply packages/oi4-oec-service-opcua-model/src/opcua/OPCUABuilder.ts]: I implemented it. but in the past the 'lastMessageID' was only updated if there wasn't any conflict between the two messages. I can't set it all the time. So I kept the update logic as it is. (see in the PR)

**Review by Weinschenk — APPROVED** [review]

**Pull request merged**